docs/source/csdl-spec.md

```diff
@@ -0,0 +1,630 @@
+- the CSDL must define certain types as described in the [mandatory types](#mandatory-types) section.
+
+### Mandatory Types
+
+A CSDL document **must** define a `csdl_Graph` enum. Each enum value defines a subgraph. Each value **must** be annotated with a `@csdl_endpoint` directive specifying the endpoint URL for the subgraph.
+
+```graphql
+enum csdl_Graph {
+  ROCKETS @csdl_endpoint(url: "https://rockets.api.com"),
+  ASTRONAUTS @csdl_endpoint(url: "https://astronauts.api.com"),
+}
+```
+
+The `csdl_Graph` enum is used as input to csdl directives which link fields and types to subgraphs.
+
+## Scalars
+
+### `scalar csdl_SelectionSet`
+
+A GraphQL selection set represented with the same syntax as defined in [the GraphQL SDL](https://spec.graphql.org/draft/#sec-Selection-Sets).
+
+Note: Unlike the federation `_FieldSet` scalar, a `csdl_SelectionSet` must parse as a GraphQL selection set, so the selections must be surrounded by braces.
+
+### `scalar csdl_Url`
+
+A [URL](https://www.w3.org/Addressing/URL/url-spec.html).
+
+### `scalar csdl_Version`
+
+A [Semantic versioning 2.0.0](https://semver.org/spec/v2.0.0.html) verison or range.
+
+## Data Model
+TK, draws heavily from federation, a little more general.
+
+### Realized types
+
+A type is *realized* wherever a resolver from a particular *subgraph* returns that type. We use the notation `subgraph::Type` to represent `Type` realized by `subgraph` and `subgraph::Type.field` to represent the resolver for `Type.field` within `subgraph`.
+
+<a class="listing">Query with resolvers and realized types</a>
+```graphql
+query {          # resolver                    | realizes
+                 # ----------------------------|--------------
+  rockets {      # rockets::Query.rockets     -> rockets::Rocket
+    captain {    # rockets::Rocket.astronaut  -> rockets::Astronaut
+      name       # astronauts::Astronaut.name -> String
+      rocket {   # rockets::Astronaut.rocket  -> rockets::Rocket
+        id       # rockets::Rocket.id         -> ID
+      }
+    }
+  }
+
+}
+```
+
+These distinctions are immaterial to the client. As far as clients are concerned, all selections on `Rocket` are the same.
+
+This illusion is maintained by the router, which must break incoming queries into subqueries. In the example above, the path `rockets.captain` is resolved by the `rockets::Rocket.astronaut` resolver, so its selection set is realized as a `rockets::Astronaut`. However, there is no `name` field for `rockets::Astronaut`—instead, the only resolver for `Astronaut.name` is `astronauts::Astronaut.name`. To process this query, the router must "convert" a `rockets::Astronaut` to an `astronauts::Astronaut` via an `_entities` query.
+
+### Portability
+A type is *portable to* a subgraph if it has a `@csdl_key` for that subgraph.
```

Comment:
nitpick: In my head, the word I gravitate toward is "defined in" rather than "portable to" — the latter implies finality, whereas the former implies temporality.

Reply:
I don't love the term portability/porting, but isn't defined in more confusing? What do you call non-federated types which are… uh… defined in a particular service, but not shared? or value types, which are *not* portable but have only free fields?